### PR TITLE
document parsing from and generating to java.lang.String

### PR DESCRIPTION
**Pull request by lmsurpre**
pretty basic info but i've had this question a few times and so thought it might be worth mentioning in the docs

also made a few more model guide tweaks/additions while I was in there (partially due to review comment)

Signed-off-by: Lee Surprenant <lmsurpre@merative.com>

### CODE REVIEW

File: docs/src/pages/guides/FHIRModelGuide.mdx [hunk 1]
```diff
@@ -76,15 +76,18 @@ bodyWeight = bodyWeight.toBuilder()
 ## Parsing a Resource from an InputStream or Reader
 
 ```
-// Parse from InputStream
+// Parse JSON from InputStream
```
Comment (punktilious) [inline docs/src/pages/guides/FHIRModelGuide.mdx:79]: Is it also worth mentioning how to configure javax.ws.rs.client.Client using providers e.g.:

```
        ClientBuilder cb =
                ClientBuilder.newBuilder()
                    .register(new FHIRProvider(RuntimeType.CLIENT))
                    .register(new FHIRJsonProvider(RuntimeType.CLIENT))
                    .register(new FHIRJsonPatchProvider(RuntimeType.CLIENT));
```

Comment (lmsurpre) [reply docs/src/pages/guides/FHIRModelGuide.mdx:79]: I'm not sure if that makes sense here or not.  I suppose we could add a section at the bottom.

Side-note:  I bet most clients just need the FHIRProvider.  FHIRJsonProvider just lets you read and write from JsonObject (which exposes a capability that our parser has to do the same)...I'm not actually sure where its used though.  FHIRJsonPatchProvider is only relevant if your issuing JSON patches.

**Review by punktilious — APPROVED** [review]
LGTM

File: docs/src/pages/guides/FHIRModelGuide.mdx (hunk 1 above)
Comment (lmsurpre) [reply docs/src/pages/guides/FHIRModelGuide.mdx:93]: ok, i added a section for it

**Pull request merged**